honeybadger-java/src/main/java/io/honeybadger/reporter/HoneybadgerReporter.java

```diff
@@ -0,0 +1,427 @@
+                return msg;
+            }
+
+            return msg.substring(0, contextSeparatorPos).trim();
+        } else {
+            return throwable.getMessage();
+        }
+    }
+
+    /**
+     * Send an error encoded in JSON to the Honeybadger API.
+     *
+     * @param jsonError Error JSON payload
+     * @return Status code from the Honeybadger API
+     * @throws IOException thrown when a network was encountered
+     */
+    private Response sendToHoneybadger(final String jsonError) throws IOException {
+        URI honeybadgerUrl = URI.create(
+                String.format("%s/%s", getConfig().getHoneybadgerUrl(), "v1/notices"));
+        Request request = buildRequest(honeybadgerUrl, jsonError);
+
+        return request.execute();
+    }
+
+    /**
+     * Builds a Apache HTTP Client request object configured for calling the
+     * Honeybadger API.
+     *
+     * @param honeybadgerUrl Endpoint location
+     * @param jsonError Error JSON payload
+     * @return a configured request object
+     */
+    private Request buildRequest(final URI honeybadgerUrl, final String jsonError) {
+        Request request = Request
+               .Post(honeybadgerUrl)
+               .addHeader("X-API-Key", getConfig().getApiKey())
+               .addHeader("Accept", "application/json")
+               .version(HttpVersion.HTTP_1_1)
+               .bodyString(jsonError, ContentType.APPLICATION_JSON);
+
+        if (System.getProperty("http.proxyHost") != null &&
+            !System.getProperty("http.proxyHost").isEmpty()) {
+            int port = Integer.parseInt(System.getProperty("http.proxyPort"));
+            HttpHost proxy = new HttpHost(System.getProperty("http.proxyHost"),
+                                          port);
+
+            request.viaProxy(proxy);
+        }
+
+        return request;
+    }
+
+    /**
+     * Tests to see if a given exception class has embedded context variables
+     * like {@link org.apache.commons.lang3.exception.ContextedException}.
+     *
+     * @param clazz class to compare
+     * @return true if a contexted exception, otherwise false
+     */
+    private static boolean exceptionClassHasContextedVariables(final Class<?> clazz) {
+        if (exceptionContextClasses == null || exceptionContextClasses.isEmpty()) {
+            return false;
+        }
+
+        for (Class<?> exceptionClass : exceptionContextClasses) {
+            if (exceptionClass.isAssignableFrom(clazz)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @return The Apache Commons Lang3 exception context class or null if not found
+     */
+    @SuppressWarnings("LiteralClassName")
+    private static Set<Class<?>> findExceptionContextClasses() {
+        final String[] classNames = new String[] {
+                "org.apache.commons.lang3.exception.ExceptionContext",
+                "com.joyent.manta.org.apache.commons.lang3.exception.ExceptionContext"
```

Comment:
We should probably delete the shaded class that I was using from testing with my monitoring utility and provide a way for users to configure their own class names here.

Reply:
I'll create a separate issue for the customization.